ignition/include/gympp/gazebo/GazeboWrapper.h

```diff
@@ -16,6 +16,10 @@
 #include <string>
 #include <vector>
 
+#include <ignition/common/URI.hh>
+#include <ignition/fuel_tools/ClientConfig.hh>
+#include <ignition/fuel_tools/FuelClient.hh>
```

Comment:
The public APIs of gym-ignition must not contain any Ignition-related headers not publicly linked Ignition library. This approach allows downstream users that want to import our C++ classes in their CMake to avoid to have also Ignition targets to handle. Think about it as a layer to isolate Ignition Robotics code from downstream users, we expose only plain C++ APIs plus classes defined internally.

This is more clear if you think of the Python bindings.  The method `myFetchResourceUri` accepts a `ignition::common::URI` and the users have no way to create this type since the URI class is not exposed to Python.

The canonical pattern that guarantees this isolation is [pimpl idiom](https://cpppatterns.com/patterns/pimpl.html). However, I understand that you wanted to make it public explicitly. Is accepting an URI a requirement? Otherwise, we can remove this URI variant and just use the string method which is compatible with the APIs we want to expose.